fluentd/configs.d/filter-pre-mux-client.conf

```diff
@@ -5,6 +5,8 @@
   shared_key "#{File.open('/etc/fluent/muxkeys/shared_key') do |f| f.readline end.rstrip}"
   secure yes
   ca_cert_path /etc/fluent/muxkeys/ca
+  buffer_queue_limit "#{ENV['BUFFER_QUEUE_LIMIT']}"
```

Comment:
> Late review, but are these okay defaults? If we have 1,024 * 16 MB chunks in the queue, how much memory does fluentd use then? Isn't that 16 GB of memory?

> Wouldn't 1 MB chunks be sufficient, with a flush_interval of 1 s to try to keep the queue short?
I agree 16 GB of memory is too large.  The default value set given by fluentd is buffer_chunk_size=8m, buffer_queue_limit=256, i.e., 2GB, in total.

Indeed, we have to make sure the buffer_chunk_limit should be able to be flushed to the ElasticSearch in flush_interval.  If flush_interval is 1 sec (1/5 of the current value), what would be the optimal size of chunk size?  1mb (1/8 of the current value)?

Probably, we should provide some formula to determine the value set of "buffer_chunk_limit, buffer_queue_limit, and flush_interval" based on the available (or specified) physical memory size?  For instance, if we choose 2 out of 3 parameters, the last one is automatically chosen not to exceed the available resource?

Another thing we'd like to consider is the fluentd parameters are shared between the client side fluentd and the mux.  The requirements for the default value set to the client side fluentd and the mux are different.  The mux should have enough capacity to collect the logs from client side fluentd's and send them to the ElasticSearch.  That's one of the motivations for making these parameters configurable.  We don't think one set does not fit all...

That is, we need to have 2 sets of default values (sorry, this is not very accurate.  Of course, the default value set is just one, but I expect we could put them in the env file in the openshift-ansible.)  And there are 3 type of use cases of fluentd, the standalone type, the client side fluentd and the mux.  (could there be something else we should be aware of?)  I think it's ideal that the standalone and the mux could share one default value set; the client side could have another which values are less than the ones in the other set to prevent overflow problem on the mux.  (I'm afraid it may add more complexity to the openshift-ansible...)

> Additionally, if we need a bigger queue, then we'll want to use the file buffer instead of memory buffer.

Currently, some customers are facing "queue size exceeds limit" with the default size 256.  We need to have some way to increase buffer_queue_limit.  Are you suggesting we should provide a method to increase buffer_queue_limit AND switch to use the file buffer?  Or we want to use the file buffer all the time?